.github/workflows/cut-rc.yml

```
@@ -23,10 +23,26 @@ env:
   jira_projects: ROX, RS, RTOOLS
   slack_channel: C03KSV3N6N8 #test-release-automation
   script_url: /repos/${{github.repository}}/contents/.github/workflows/scripts/common.sh?ref=${{github.event.repository.default_branch}}
-  DRY_RUN: ${{ fromJSON('["true", "false"]')[inputs.dry-run != 'true'] }}
+  DRY_RUN: ${{ fromJSON('["true", "false"]')[github.event.inputs.dry-run != 'true'] }}
```

Could you maybe explain this a bit? Do you store the negative value of `inputs.dry-run` in `DRY_RUN`?

Could we think of a positive logic here - i.e., using `=` instead of `!=`?

For example, would this be possible?

```suggestion
  DRY_RUN: ${{ fromJSON('["true", "false"]')[github.event.inputs.dry-run = 'true'] }}
```

---

`dry-run` is passed as a string with 3 possible values: 'true', 'false' or ''.
'false' or 'empty' are functionally equivalent, meaning it is a wet run, not a dry one.
Since we don't have much freedom in ${{ }} expressions, the trick is to use JSON list indexes (0 or 1) to evaluate conditions.

The logic here is to have either `'true'` or `'false'` in `DRY_RUN`, and never have it empty.

`dry-run != 'true'` evaluates to 0 for a dry and to 1 otherwise. Therefore `DRY_RUN` holds `'true'` for 0 and `'false'` for 1, essentially copying the value from `dry-run` and evaluating `'false'` if `dry-run` is empty.

Oh, hope this helps!